pkg/routeagent_driver/handlers/kubeproxy/routes_iface.go

```diff
@@ -72,20 +73,23 @@ func (kp *SyncHandler) updateRoutingRulesForCIDRBlock(inputCidrBlock string, ope
 
 	switch operation {
 	case Add:
-		if kp.routeCacheGWNode.Add(inputCidrBlock) {
+		if !kp.routeCacheGWNode.Has(inputCidrBlock) {
 			if err := kp.configureRoute(inputCidrBlock, operation, viaGW); err != nil {
-				kp.routeCacheGWNode.Remove(inputCidrBlock)
 				logger.Errorf(err, "Failed to add route %q for HostNetwork support on the Gateway node: %v",
 					inputCidrBlock, err)
 			}
+
+			kp.routeCacheGWNode.Insert(inputCidrBlock)
 		}
 
 	case Delete:
-		if kp.routeCacheGWNode.Remove(inputCidrBlock) {
+		if kp.routeCacheGWNode.Has(inputCidrBlock) {
 			if err := kp.configureRoute(inputCidrBlock, operation, viaGW); err != nil {
 				logger.Errorf(err, "Failed to delete route %q for HostNetwork support on the Gateway node",
 					inputCidrBlock)
 			}
+
+			kp.routeCacheGWNode.Delete(inputCidrBlock)
```

Comment:
The prior behavior was to remove `inputCidrBlock` from the set regardless of an error on L87 so, to preserve that, this line should be before L87.

Reply:
Right, same when adding it, I’ll fix both.